src/contrib/tests/system-tests.rst

```diff
@@ -6,6 +6,173 @@ System Tests
+
+* **test**\_ Pytest discovery pattern.
+* test\_ **filename**\__ File name, no extension and a double underscore separating the file name from the test name.
+* test_filename\__ **name_of_the_test** Test name (see below).
+
+Naming criteria
+'''''''''''''''''
+
+* Test names should describe the expected behavior of what you are testing.
+* Test names should describe what the code is doing.
+* Test names should provide a good idea of what the test does.
+* Test names can use the filename to help describe what the test does.
+
+Docstring
+'''''''''''
+
+All tests require the following documentation strings.
+
+* **name**: Test name without the_under_scores.
+* **description**: If the description is too long, a new line, indented, is shown below. Optional, if the name is not clear enough. Any additional information that helps others understand the purpose of the test.
+* **setup**: All steps leading up to the scope of the test.
+* **steps**: Only relevant steps that are described in the test name.
+* **expectedresults**: Test results must equal the number of steps.
+* **customerscenario**: True, if this is a customer scenario.
+
+The priority of the docstring is to understand the purpose of the test and provide any information that will help future maintainability.
+The docstring will provide more of a summary rather than specific details of the test. In a well-written test, the details will be easily readable from the test code.
+
+.. code-block::
+    :caption: docstring example
+
+     """
+     :title: Feature presence
+     :description:
+         The parametrization states the distribution name, distribution version, SSSD version and feature
+         presence.
+         As an example, ("Fedora", 39, 0, 2, 9, "passkey", True) should be read in the following way:
+         In a Fedora 39 or higher system with SSSD 2.9 or higher, passkey feature shall be present.
+         Another example, (None, None, None, 2, 10, "knownhosts", True):
+         In a system with SSSD 2.10 or higher, knownhosts feature shall be present.
+     :setup:
+         1. Skip if distribution name doesn't match
+         2. Skip if distribution version doesn't match
+     :steps:
+         1. Check SSSD version and feature presence
+     :expectedresults:
+         1. Depending on the parameterization, the feature shall be present or not
+     :customerscenario: False
+     """
+
+Code
+^^^^^^
+
+While the docstrings contain the emphasis of what and why. The code will contain how. There is a strong preference making the code readable and efficient over anything else like making the test more comprehensive, elegant or concise. There is no applicable rule but suggestions. If it's complicated, maybe split into a couple of test cases. If reading code feels like a tennis match bouncing back and forth between parts of the code and docstrings, it can be improved. Take the following.
```

Comment:
I'm not completely sure, but I think:
The code will contain how -> The code will contain **the** how

In addition:
Take the following. -> Take the following **example**.